Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -414,53 +414,91 @@ public DateTime Value
         }
 
-        // Ensure to handle DST transitions correctly when timezones are involved.
-        var newDateTime = TzId is null ? Value.Add(ts) : AsUtc.Add(ts);
+        // RFC 5545 3.3.6:
+        // If the property permits, multiple "duration" values are specified by a COMMA-separated
+        // list of values.The format is based on the[ISO.8601.2004] complete representation basic
+        // format with designators for the duration of time.The format can represent nominal
+        // durations(weeks and days) and accurate durations(hours, minutes, and seconds).
+        // Note that unlike[ISO.8601.2004], this value type doesn't support the "Y" and "M"
+        // designators to specify durations in terms of years and months.
+        //
+        // The duration of a week or a day depends on its position in the calendar. In the case
+        // of discontinuities in the time scale, such as the change from standard time to daylight
+        // time and back, the computation of the exact duration requires the subtraction or
+        // addition of the change of duration of the discontinuity.Leap seconds MUST NOT be
+        // considered when computing an exact duration.When computing an exact duration, the
+        // greatest order time components MUST be added first, that is, the number of days MUST be
+        // added first, followed by the number of hours, number of minutes, and number of seconds.
+
+        (TimeSpan? nominalPart, TimeSpan? exactPart) dt;
+        if (TzId is null)
+            dt = (d.ToTimeSpan(), null);
+        else
+            dt = (d.HasDate ? d.DateAsTimeSpan : null, d.HasTime ? d.TimeAsTimeSpan : null);
+
+        var newDateTime = this;
+        if (dt.nominalPart is not null)
+            newDateTime = new CalDateTime(newDateTime.Value.Add(dt.nominalPart.Value), TzId, HasTime);
 
-        var copy = Copy<CalDateTime>();
-        copy._dateOnly = DateOnly.FromDateTime(newDateTime);
-        copy._timeOnly = HasTime ? TruncateTimeToSeconds(newDateTime) : null;
-        copy._tzId = TzId is null ? null : UtcTzId;
+        if (dt.exactPart is not null)
+            newDateTime = new CalDateTime(newDateTime.AsUtc.Add(dt.exactPart.Value), UtcTzId, HasTime);
+        
+        if (TzId is not null)
+            // Convert to the original timezone even if already set to ensure we're not in a non-existing time.
+            newDateTime = newDateTime.ToTimeZone(TzId);
 
-        return TzId is null ? copy : copy.ToTimeZone(TzId);
+        AssociateWith(this);
+
+        return newDateTime;
     }
 
-    /// <summary>Returns a new <see cref="TimeSpan" /> from subtracting the specified <see cref="IDateTime"/> from to the value of this instance.</summary>
-    /// <param name="dt"></param>
-    public TimeSpan Subtract(IDateTime dt) => AsUtc - dt.AsUtc;
+    /// <inheritdoc/>
+    public TimeSpan SubtractExact(IDateTime dt) => AsUtc - dt.AsUtc;
```

Comment:
A xmldoc hint for SubtractExact vs Subtract similar like in static Duration FromTimeSpan(TimeSpan t) might be useful.